models.js

```diff
@@ -18,4 +18,18 @@ exports.fetchArticles = () => {
     return articles.rows
   })
 }
-};
+
+exports.fetchArticleById = (article_id) => {
+  const acceptedInput = new RegExp(/^\d+(?:\.\d{1,2})?$/
+)
+  if(acceptedInput.test(article_id) === false) {
+    return Promise.reject({status: 400, msg: 'bad article request'})
+  }
+  return db.query(`SELECT * FROM articles WHERE article_id = ${article_id}`)
+  .then((article)=> {
+    if(article.rows.length < 1){ 
+      return Promise.reject({status: 404, msg: 'article not found'})
+    }
+    return article.rows.pop()
```

Comment:
this is a fun little pop ... don't think I've seen this done on BE project before 👍